packages/formstr-app/src/containers/CreateForm/components/FormSettings/RelayList.tsx

```diff
@@ -0,0 +1,206 @@
+import { Button, Collapse, Input, Typography } from "antd";
+import { getDefaultRelays } from "@formstr/sdk";
+import { makeTag } from "@formstr/sdk/dist/utils/utils";
+import {
+  CloseOutlined,
+  DeleteOutlined,
+  EditOutlined,
+  SaveOutlined,
+} from "@ant-design/icons";
+import { ChangeEvent, useState } from "react";
+import useFormBuilderContext from "../../hooks/useFormBuilderContext";
+
+const { Text } = Typography;
+
+function isValidWebSocketUrl(url: string): boolean {
+  const websocketUrlPattern =
+    /^(wss?:\/\/)([^:@/]+(?::[^@/]+)?@)?([^:/]+)(?::(\d+))?(\/.*)?$/;
+
+  const match = url.match(websocketUrlPattern);
+
+  if (!match) {
+    return false;
+  }
+
+  const [, scheme, , , port] = match;
+
+  if (!scheme || (scheme !== "ws://" && scheme !== "wss://")) {
+    return false;
+  }
+  if (port !== undefined) {
+    const portNumber = parseInt(port, 10);
+    if (!(0 <= portNumber && portNumber <= 65535)) {
+      return false;
```

Comment:
do we plan to give specific error
eg. port invalid/scheme invalid?

Reply:
not really